sw/device/lib/base/macros.h

```diff
@@ -269,6 +270,164 @@ extern "C++" {
+#define OT_VA_FOR_EACH_13_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_12_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_14_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_13_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_15_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_14_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_16_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_15_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_17_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_16_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_18_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_17_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_19_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_18_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_20_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_19_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_21_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_20_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_22_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_21_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_23_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_22_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_24_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_23_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_25_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_24_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_26_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_25_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_27_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_26_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_28_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_27_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_29_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_28_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_30_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_29_(action, ##__VA_ARGS__)
+#define OT_VA_FOR_EACH_31_(action, item, ...) \
+  action(item);                               \
+  OT_VA_FOR_EACH_30_(action, ##__VA_ARGS__)
+
+/**
+ * A macro that checks whether a specified argument is not a standard C integer
+ * type with a width greater than 32 bits. Useful when assuming that variable
+ * arguments are 32 bits integers.
+ *
+ * @param arg An argument/expression
+ */
+#if ULONG_LONG_MAX != UINT64_MAX
```

Comment:
I think these are never hit, regardless 32-bit or 64-bit.

Reply:
That makes sense for 32-bit and 64-bit. I added this to account for a hypothetical 128-bit case where `long long` is not defined to be the same size as a 64 bit integer, but I'll opt to remove this for now for coverage related reasons as this branch will indeed never be hit by current systems.